neutrons/SNAPRed
# refactor of diff cal with new mantid changes

**Description of work**

These algorithms were originally implemented in 
 - [PR #53 ](https://github.com/neutrons/SNAPRed/pull/53)
 - [PR #72](https://github.com/neutrons/SNAPRed/pull/72)
 
They were implemented as part of the following stories:
- [EWM 2166](https://ornlrse.clm.ibmcloud.com/ccm/web/projects/Neutron%20Data%20Project%20%28Change%20Management%29#action=com.ibm.team.workitem.viewWorkItem&id=2166)
- [EWM 1965](https://ornlrse.clm.ibmcloud.com/ccm/web/projects/Neutron%20Data%20Project%20%28Change%20Management%29#action=com.ibm.team.workitem.viewWorkItem&id=1965)
- [EWM 1126](https://ornlrse.clm.ibmcloud.com/ccm/web/projects/Neutron%20Data%20Project%20%28Change%20Management%29#action=com.ibm.team.workitem.viewWorkItem&id=1965)

Since then, work was completed on the following story in mantid:
- [EWM 1851](https://ornlrse.clm.ibmcloud.com/ccm/web/projects/Neutron%20Data%20Project%20%28Change%20Management%29#action=com.ibm.team.workitem.viewWorkItem&id=1851)
- [EWM 2350](https://ornlrse.clm.ibmcloud.com/ccm/web/projects/Neutron%20Data%20Project%20%28Change%20Management%29#action=com.ibm.team.workitem.viewWorkItem&id=2350)
- [Mantid PR #35862](https://github.com/mantidproject/mantid/pull/35862)
- [Mantid PR #35896](https://github.com/mantidproject/mantid/pull/35896)

The above allow simpler use of the `ConvertDiffCal` and `CalculateDIFC` algorithms, along with safer use of `Rebin` in logarithmic binning.

This PR is meant to update the algorithms/recipe in Diffraction Calibration to use these new mantid features.

It is also meant to fix some small defects that have come up during CIS testing in the hackathon:
- [EWM 2335](https://ornlrse.clm.ibmcloud.com/ccm/web/projects/Neutron%20Data%20Project%20(Change%20Management)#action=com.ibm.team.workitem.viewWorkItem&id=2335)

**To test:**

Firstly, run this in pytest to make sure all tests pass.

In mantid workbench, open the original prototype script.  Replace all of the calls to `ConvertDiffCal` and `CalculateDIFC` to use `OffsetMode = "Signed"` and provide the `BinWidth`.  Also convert all of the `Rebin` to use `BinningMode="Logarithmic"`.  Make sure the script still runs.

In mantid workbench, create a set of `DiffractionCalibrationIngredients`.  With these ingredients, call the algorithms `CalculateOffsetDIFC` and `GroupByGroupCalibration` and verify they run and properly process output data.  Also call the recipe `DiffractionCalculationRecipe` with these ingredients, and verify everything runs and produces properly processed output.

**Link to EWM item**
see above
